examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,271 @@
+    for filename, nodename, linear_threshold in [
+        (
+            "240819_test_amp0p05_calib_pulse_calib.h5",
+            "pulse_calib_3",
+            150e-6,
+        ),  # low power
+        (
+            "240819_amp0p1_calib_pulse_calib.h5",
+            "pulse_calib_1",
+            270e-6,
+        ),  # low power
+        (
+            "240819_amp0p2_calib_repeat_pulse_calib.h5",
+            "pulse_calib_8",
+            310e-6,
+        ),  # low power
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        assert (
+            d.get_prop("postproc_type") == "GDS_capture_v1"
+        ), "The wrong postproc_type was set so you most likely used the wrong script for acquisition"
+        amplitude = d.get_prop("acq_params")["amplitude"]
+        fl.basename = f"amplitude = {amplitude}"
+        if not d.get_units("t") == "s":
+            print(
+                "units weren't set for the t axis or else I can't read them from the hdf5 file!"
+            )
+            d.set_units("t", "s")
+        d *= V_atten_ratio
+        d /= np.sqrt(50)  # V/sqrt(R) = sqrt(P)
+
+        def switch_to_plot(d, j):
+            thislen = d["t_pulse"][j]
+            fl.next(f"pulse length = {thislen}")
+
+        def indiv_plots(d, thislabel, thiscolor):
+            if skip_plots is None:
+                return
+            for j in range(len(d["t_pulse"])):
+                if j % skip_plots == 0:
+                    switch_to_plot(d, j)
+                    fl.plot(
+                        d["t_pulse", j],
+                        alpha=0.2,
+                        color=thiscolor,
+                        label=thislabel,
+                    )
+
+        indiv_plots(d, "raw", "blue")
+        # {{{ data is already analytic, and downsampled to below 24 MHz
+        indiv_plots(abs(d), "analytic", "orange")
+        d.ft("t")
+        # {{{ apply frequency filter
+        d.ift("t")
+        SW = 1 / (d["t"][1] - d["t"][0])  # sample rate
+        carrier = (
+            d.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+        )  # signal frequency
+        n = np.floor(
+            (carrier + SW / 2) / SW
+        )  # nearest integer multiple of sampling frequency
+        nu_a = carrier - n * SW  # aliasing frequency
+        if nu_a < 0:
+            perceived_nu = abs(nu_a)
+            # we removed the negative frequencies so it's counterpart is the abs
+            d.run(np.conj)
+        else:  # otherwise subtract the difference between the signal and nyquist from nyquist's
+            perceived_nu = SW - nu_a
+        d.ft("t")
+        d["t" : (0, perceived_nu - 0.5 * slicewidth)] *= 0
+        d["t" : (perceived_nu + 0.5 * slicewidth, None)] *= 0
+        # }}}
+        d.ift("t")
+        indiv_plots(abs(d), "filtered analytic", "red")
+        fl.next("collect filtered analytic", legend=True)
+        for j in range(d.shape["t_pulse"]):
```

Comment:
It seems like this for block mostly does the same thing that you do further down. Why not just move this code in with the stuff down below

Reply:
And remove the redundancy

Reply:
I also don't understand what you are doing here that is not handled by the indiv_plots function. 

Overall, it's very likely that the code needs to be cleaned up, or else you need in line code comments to explain what is going on

Reply:
knowing this is rhetorical I've added comments in the script for clarification